src/utils/hsdis/binutils/hsdis-binutils.c

```diff
@@ -242,11 +242,38 @@ static const char* format_insn_close(const char* close,
   case dis_noninsn:     type = "noninsn";    break;
   }
 
-  strcpy(buf, close);
-  char* p = buf;
-  if (type)    sprintf(p += strlen(p), " type='%s'", type);
-  if (dsize)   sprintf(p += strlen(p), " dsize='%d'", dsize);
-  if (delays)  sprintf(p += strlen(p), " delay='%d'", delays);
+  size_t used_size = snprintf(buf, bufsize, "%s", close);
+  if ((used_size < 0) || (used_size >= bufsize)) {
```

Comment:
(used_size < 0) is tautologically false, since used_size is a size_t, so unsigned.  I'm somewhat surprised
this doesn't trigger a warning from some compiler.

Reply:
bufsize is size_t, so that's a comparison between signed and unsigned values, which I think some compilers
will warn about.  Maybe the preceding check for negative is getting rid of that?  But will that still occur in
a slowdebug build, or will the lack of optimization lead to a warning?

Reply:
As always, this comment helps a lot.  Thank you!

Updated to cast `int` to `size_t` explicitly.

Reply:
@kimbarrett Did you have a chance to have another look?  Please let me know if you prefer to the update that the returned value of snprintf() is not checked because the memory size has been checked previously.